### PR TITLE
Use Geotrellis to produce TR-55 Input. 

### PR DESCRIPTION
**Pull request by jamesmcclain**
Connects #585

**To Test**

   1. Do all of the things needed to run https://github.com/WikiWatershed/model-my-watershed/pull/589
   2. Reprovision `app` and `worker`, or manually update the TR-55 code on those two VMs (`sudo pip uninstall tr55` then `sudo pip install 'git+https://github.com/jamesmcclain/tr-55.git'`).

**Caveats**

   1. Many of the caveats which apply to https://github.com/WikiWatershed/model-my-watershed/pull/589 also apply here, in particular the small size of the test area.
   2. Do not overlap reclassifications and/or BMPs.  There is no fundamental reason why these cannot overlap on the user's screen, but they must be provided to the code as a non-intersecting collection, which the current front-end code does not do.  At present, overlaps cause undefined results.
   3. Certain BMPS like "infiltration basins" and "porous paving" are virtually unusable because they are restricted to certain soil types.  Since you cannot tell which areas are allowed and which aren't chances are using either of those BMPs will raise an error.
   4. The water quality results may or may not be valid: those calculations need to know the area of a tile, and the value currently being used may not be correct.

screenshot from 2015-07-23 15 13 36

screenshot from 2015-07-23 15 13 06

screenshot from 2015-07-23 15 14 03

screenshot from 2015-07-23 15 14 32

### CODE REVIEW

**Comment by jamesmcclain** [comment]
I added some new tests which pass locally but fail on Jenkins.
```
======================================================================
ERROR: test_census (app.apps.modeling.tests.ExerciseGeoprocessing)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/opt/app/apps/modeling/tests.py", line 54, in test_census
    actual = geoprocessing.geojson_to_census(self.polygon)
  File "/opt/app/apps/modeling/geoprocessing.py", line 103, in geojson_to_census
    data = json.loads(geotrellis(polygon))
  File "/usr/lib/python2.7/json/__init__.py", line 338, in loads
    return _default_decoder.decode(s)
  File "/usr/lib/python2.7/json/decoder.py", line 366, in decode
    obj, end = self.raw_decode(s, idx=_w(s, 0).end())
TypeError: expected string or buffer

======================================================================
ERROR: test_geotrellis_good (app.apps.modeling.tests.ExerciseGeoprocessing)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/opt/app/apps/modeling/tests.py", line 41, in test_geotrellis_good
    self.assertEqual(json.loads(actual), json.loads(expected), "Discrepant answer from Geotrellis")  # noqa
  File "/usr/lib/python2.7/json/__init__.py", line 338, in loads
    return _default_decoder.decode(s)
  File "/usr/lib/python2.7/json/decoder.py", line 366, in decode
    obj, end = self.raw_decode(s, idx=_w(s, 0).end())
TypeError: expected string or buffer

======================================================================
ERROR: test_survey (app.apps.modeling.tests.ExerciseGeoprocessing)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/opt/app/apps/modeling/tests.py", line 49, in test_survey
    actual = geoprocessing.geojson_to_survey(self.polygon)
  File "/opt/app/apps/modeling/geoprocessing.py", line 173, in geojson_to_survey
    data = json.loads(geotrellis(polygon))
  File "/usr/lib/python2.7/json/__init__.py", line 338, in loads
    return _default_decoder.decode(s)
  File "/usr/lib/python2.7/json/decoder.py", line 366, in decode
    obj, end = self.raw_decode(s, idx=_w(s, 0).end())
TypeError: expected string or buffer

======================================================================
ERROR: test_tr55_job_runs_in_chain (app.apps.modeling.tests.TaskRunnerTestCase)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/local/lib/python2.7/dist-packages/django/test/utils.py", line 196, in inner
    return test_func(*args, **kwargs)
  File "/opt/app/apps/modeling/tests.py", line 111, in test_tr55_job_runs_in_chain
    self.job.id)
  File "/opt/app/apps/modeling/views.py", line 215, in _initiate_tr55_job_chain
    return chain(job_chain).apply_async(link_error=save_job_error.s(job_id))
  File "/usr/local/lib/python2.7/dist-packages/celery/canvas.py", line 251, in apply_async
    return _apply(args, kwargs, **options)
  File "/usr/local/lib/python2.7/dist-packages/celery/app/builtins.py", line 275, in apply_async
    return self.apply(args, kwargs, **options)
  File "/usr/local/lib/python2.7/dist-packages/celery/app/builtins.py", line 302, in apply
    last and (last.get(), ),
  File "/usr/local/lib/python2.7/dist-packages/celery/result.py", line 870, in get
    raise self.result
TypeError: expected string or buffer

======================================================================
FAIL: test_geotrellis_bad (app.apps.modeling.tests.ExerciseGeoprocessing)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/opt/app/apps/modeling/tests.py", line 46, in test_geotrellis_bad
    self.assertEqual(actual, expected, "Discrepant answer from Geotrellis")
AssertionError: Discrepant answer from Geotrellis
```

**Comment by lliss** [comment]
I did review the code and it appears fine to me. +1 there. All tests continue to pass.

File: src/mmw/apps/modeling/geoprocessing.py [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+import httplib
+import json
+
+
+NLCD_MAPPING = {
+    11: ['water', 'Water'],
+    21: ['urban_grass', 'Urban- or Tall-Grass'],
+    22: ['li_residential', 'Low-Intensity Res.'],
+    23: ['hi_residential', 'High-Intensity Res.'],
+    24: ['industrial', 'Industrial &c.'],
```
Comment (rajadain) [inline src/mmw/apps/modeling/geoprocessing.py]: Out of curiosity, what does `&c.` stand for? Etcetera?

Comment (rajadain) [inline src/mmw/apps/modeling/geoprocessing.py]: [Indeed it is!](http://designshack.net/articles/typography/why-i-love-ampersands-you-should-too/) Although in the interest of usability, I would recommend considering the [more common](https://en.wikipedia.org/wiki/Ampersand#Usage) representation "etc."

(The [bikeshedding](https://en.wikipedia.org/wiki/Parkinson%27s_law_of_triviality) nature of this comment is not lost on me)

Comment (jamesmcclain) [inline src/mmw/apps/modeling/geoprocessing.py]: Okay, I will take that under advisement!

**Pull request closed without merging**